src/main/java/games/strategy/engine/framework/startup/login/ClientLoginValidator.java

```diff
@@ -91,6 +115,26 @@ public String verifyConnection(final Map<String, String> propertiesSentToClient,
       return YOU_HAVE_BEEN_BANNED;
     }
     if (propertiesSentToClient.get(PASSWORD_REQUIRED_PROPERTY).equals(Boolean.TRUE.toString())) {
+      final String base64String = propertiesReadFromClient.get(ClientLogin.ENCRYPTED_PASSWORD_PROPERTY);
+      if (base64String != null) {
+        final String publicKey = propertiesSentToClient.get(RANDOM_RSA_PUBLIC_KEY_PROPERTY);
+        try {
+          final Cipher cipher = Cipher.getInstance(RSA_ECB_OAEPP);
+          cipher.init(Cipher.DECRYPT_MODE, rsaKeyMap.get(publicKey));
+          if (MessageDigest.isEqual(cipher.doFinal(Base64.getDecoder().decode(base64String)), m_password.getBytes())) {
```

Comment:
Same suggestion as above regarding explicitly specifying a `Charset` to `String#getBytes()`.

Also, would `Arrays#equals()` not work here?  Just curious why `MessageDigest#isEqual()` is being used.